source/common/vue/window/toolbar-controls/Button.vue

```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="toolbar-group">
+  <div v-if="control.visible !== false" class="toolbar-group">
```

Comment:
Please do not do this. Basically, this will mount the corresponding component but hide it completely, which wastes unnecessary resources, and might even give us performance/memory issues if we ever use these components in larger amounts.

Instead, what you want to do is have a look at the `WindowToolbar.vue` component (location: `source/common/vue/window`). In there you will see a `v-if="item.type === <some type>"` directive. This will basically only render the component whose type matches. I.e. if a toolbar control has type `button` it will basically mount a new `Button.vue`-component and then provide the settings in the `item` to that Button component instance.

What you want to also add in there is a check whether the `visible` property is false, i.e. in total, for example, you would end up with `v-if="item.type === 'button' && item.visible !== false"` — but for every possible toolbar component.

Reply:
Thanks for your guidance. My original approach was equivalent to loading and then hiding. This is really unnecessary. I will changed them as you said.